components/my-account/User/UserTable/UserTable.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+
+import { Edit as EditIcon, RemoveCircle as RemoveCircleIcon } from '@mui/icons-material'
+import {
+  Box,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Typography,
+  styled,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material'
+import { useTranslation } from 'next-i18next'
+
+import UserForm from '../UserForm/UserForm'
+
+import { B2BUser, Maybe } from '@/lib/gql/types'
+
+interface UserTableProps {
+  b2bAccountUsers: B2BUser[] | undefined
+  deleteUser: (id: Maybe<string> | undefined) => void
+  editUserId: Maybe<string> | undefined
+  setEditUserId: (id: Maybe<string> | undefined) => void
+}
+
+const UserStatusCircle = styled('div', {
+  shouldForwardProp: (prop) => prop == 'color',
+})<{ color: string }>(({ color }) => ({
+  height: '12px',
+  width: '12px',
+  left: 0,
+  top: '3.5px',
+  borderRadius: '100px',
+  marginRight: '8px',
+  background: color,
+}))
+
+const style = {
+  emailAddressCell: {
+    flex: { md: 1, xs: 0.5 },
+    maxWidth: { xs: '150px', md: '100%' },
+    overflow: { xs: 'hidden' },
+    textOverflow: { xs: 'ellipsis' },
+  },
+}
+
+const UserTable = (props: UserTableProps) => {
+  const { b2bAccountUsers, deleteUser, editUserId, setEditUserId } = props
+
+  const { t } = useTranslation('common')
+  const theme = useTheme()
+  const mdScreen = useMediaQuery(theme.breakpoints.up('md'))
+
+  return (
+    <Table style={{ minHeight: b2bAccountUsers && b2bAccountUsers.length ? 0 : '345px' }}>
+      <TableHead>
+        <TableRow style={{ backgroundColor: theme.palette.grey[100] }}>
+          <TableCell
+            colSpan={2}
+            sx={{ flex: 1, width: { xs: '150px' }, overflow: { xs: 'hidden' } }}
+          >
+            {mdScreen ? t('email-address') : t('email')}
+          </TableCell>
+          {mdScreen && <TableCell>{t('first-name')}</TableCell>}
```

Comment:
Can we do the mdScreen check at one place and show accordingly ?

Reply:
Done